sdk/eventhub/event-hubs/src/eventHubSender.ts

```diff
@@ -508,164 +498,202 @@ export class EventHubSender extends LinkEntity {
+            this._sender.removeListener(SenderEvents.released, onReleased);
+            this._sender.removeListener(SenderEvents.modified, onModified);
+          }
+        };
+
+        const rejectOnSendError = (err: Error) => {
+          err = translate(err);
+          log.error(
+            "[%s] An error occurred while performing send on %s",
+            this._context.connectionId,
+            this.name,
+            err
+          );
+          reject(err);
+        };
+
+        const actionAfterTimeout = () => {
+          clearTimeout(waitTimer);
+          removeListeners();
+          const desc: string =
+            `[${this._context.connectionId}] Sender "${this.name}" with ` +
+            `address "${this.address}", was not able to send the message right now, due ` +
+            `to operation timeout.`;
+          log.error(desc);
+          const e: Error = {
+            name: "OperationTimeoutError",
+            message: desc
           };
-          onReleased = (context: EventContext) => {
-            removeListeners();
-            log.error(
-              "[%s] Sender '%s', got event released.",
-              this._context.connectionId,
-              this.name
-            );
-            let err: Error;
-            if (context!.delivery!.remote_state!.error) {
-              err = translate(context!.delivery!.remote_state!.error);
-            } else {
-              err = new Error(
-                `[${this._context.connectionId}] Sender '${this.name}', ` +
-                  `received a release disposition.Hence we are rejecting the promise.`
-              );
+          return reject(translate(e));
+        };
+
+        const sendEventPromise = () =>
+          new Promise<void>((resolve, reject) => {
+            if (abortSignal && abortSignal.aborted) {
+              // operation has been cancelled, so exit quickly
+              return rejectOnAbort();
             }
-            log.error(err);
-            reject(err);
-          };
-          onModified = (context: EventContext) => {
-            removeListeners();
-            log.error(
-              "[%s] Sender '%s', got event modified.",
+
+            log.sender(
+              "[%s] Sender '%s', credit: %d available: %d",
               this._context.connectionId,
-              this.name
+              this.name,
+              this._sender!.credit,
+              this._sender!.session.outgoing.available()
             );
-            let err: Error;
-            if (context!.delivery!.remote_state!.error) {
-              err = translate(context!.delivery!.remote_state!.error);
+            if (this._sender!.sendable()) {
+              log.sender(
+                "[%s] Sender '%s', sending message with id '%s'.",
+                this._context.connectionId,
+                this.name,
+                (Buffer.isBuffer(message) ? tag : message.message_id) || tag || "<not specified>"
```

Comment:
Looks like you're missing the updates from the create batch work @ShivangiReja did. Can you remerge from master again?

Reply:
Oh, not too sure why/what happened - can you guys take a look at latest version and see if everything looks right with the Git merge? @ShivangiReja

Reply:
I believe the main change is around the fact that we are no longer using the `tag`.

We have been passing the message id as tag. rhea uses it as a property on the `delivery` object that it creates.
Since in track 2 we no longer allow users to set the message id, this is no longer relevant, and therefore have been removed with @ShivangiReja's changes